components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ExportUtils.java

```diff
@@ -506,150 +502,91 @@ public static void addWSDLtoArchive(String archivePath, APIIdentifier apiIdentif
+     * @param registry     Current tenant registry
+     * @param mediationName     Name of the mediation policy
+     * @param apiIdentifier     API Identifier
+     * @throws RegistryException If an error occurs while retrieving registry elements
+     * @throws APIManagementException If an error occurs while writing the mediation policy to file
      */
-    private static void writeSequenceToFile(String pathToExportedSequence,
-            AbstractMap.SimpleEntry<String, OMElement> sequenceDetails, APIIdentifier apiIdentifier)
-            throws APIImportExportException {
-
-        if (sequenceDetails != null) {
-            String sequenceFileName = sequenceDetails.getKey();
-            OMElement sequenceConfig = sequenceDetails.getValue();
-            CommonUtil.createDirectory(pathToExportedSequence);
-            String exportedSequenceFile = pathToExportedSequence + sequenceFileName;
-            try (OutputStream outputStream = new FileOutputStream(exportedSequenceFile)) {
-                sequenceConfig.serialize(outputStream);
-                if (log.isDebugEnabled()) {
-                    log.debug(sequenceFileName + " of API: " + apiIdentifier.getApiName() + " retrieved successfully");
+    private static void writeSequenceToArchive(Resource mediationResource, String individualSequenceExportPath,
+            Registry registry, String mediationName, APIIdentifier apiIdentifier)
+            throws RegistryException, APIManagementException {
+        if (mediationResource != null) {
+            // Get the registry resource path
+            String resourcePath = mediationResource.getPath();
+            // Check whether resource exists in the registry
+            if (registry.resourceExists(resourcePath)) {
+                Resource mediationFile = registry.get(resourcePath);
+                try (OutputStream outputStream = new FileOutputStream(
+                        individualSequenceExportPath + File.separator + mediationName + APIConstants.DOT
+                                + APIConstants.XML_DOC_EXTENSION);
+                        InputStream fileInputStream = mediationFile.getContentStream()) {
+                    IOUtils.copy(fileInputStream, outputStream);
+                } catch (IOException e) {
+                    throw new APIManagementException("Error while writing the mediation sequence"+ mediationName+ "to file");
                 }
-            } catch (IOException e) {
-                throw new APIImportExportException("Unable to find file: " + exportedSequenceFile, e);
-            } catch (XMLStreamException e) {
-                throw new APIImportExportException("Error while processing XML stream ", e);
+            } else {
+                // Log error and avoid throwing as we give capability to export an API without sequences
+                log.error("Sequence resource for API/API Product: " + apiIdentifier.getName() + " not found in "
```

Comment:
Should this be logged as an error since this is a valid scenario?

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/9480/commits/d1a9950578cb6e9db87036e285e0448a53bebb42